File: Python/permute.py
import random

def swap(array, i, j):
    array[i], array[j] = array[j], array[i]

def permute(array, n):
    for i in range(n - 1, -1, -1):
        j = random.randint(0, i)
        swap(array, i, j)

array = [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 ]
permute(array, len(array))
print(array)

File: Dart/permute.dart
import 'dart:math';

void swap(List<int> A, int i, int j){
    var temp = A[i];
    A[i] = A[j];
    A[j] = temp;
}

void permute( List<int> A, int n) {
    Random random = Random();
    for ( int i = n; i > 0; i-- ) {
        int j = random.nextInt(i);
        swap(A, i - 1, j);
    }
}

void main() {
    List<int> array = [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 ];
    print("Before : $array");
    permute(array, array.length);
    print("After : $array");
}